bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -405,7 +405,27 @@ public boolean resumeSuspendedClients(final int numberOfInMemBlockPermits)
 
   public boolean isMemoryBlockAvailable()
   {
-    return (storage == null) || (numberOfInMemBlockPermits.get() > 0);
+    return (numberOfInMemBlockPermits.get() > 0) || ((storage == null) && !areSubscribersBehindByMax());
```

Comment:
Also, I think storage == null is fine here for now as it shows the two cases, spooling enabled and disabled, to be read as (storage != null) && (numberOfInMemBlockPermits.get() > 0) || (storage == null) ... Eventually when the parameter comes in these would probably go away and handled in a uniform way.